### PR TITLE
Updated user-guide and convert code RFC #176 imports

### PR DESCRIPTION
**Pull request by aureliosaraiva**
I reviewed the code samples for the ES6 format.

I reviewed the code samples for the ES6 format, I also updated the code to follow the RFC [#176](https://github.com/emberjs/rfcs/blob/master/text/0176-javascript-module-api.md)

https://github.com/ember-cli/ember-rfc176-data

**Revised**
- [x] Getting started
- [x] Usign modules & the resolver
- [x] Naming conventions
- [x] Usign with ember data
- [x] Testing
- [x] Asset compilation
- [x] managing dependencies
- [x] Enviroments
- [x] Deployments
- [x] Upgrading
- [x] common issues
- [x] Windows
- [x] Editors
- [x] Tutorials
- [ ] Extends
- [ ] Developing addons and blueprints

**Improve asset documentation**
https://ember-cli.com/user-guide/#raw-assets

### CODE REVIEW

**Review by jenweber — APPROVED** [review]
So far, so good! I didn't spot any issues. I reviewed the diff and scanned the built app for any visual bugs. Good catch on base/rootURL. I think at least 1 other person should review before merge. Awesome work 🎉

**Review by locks — APPROVED** [review]
Excellent work, thanks for the help!

**Pull request merged**
